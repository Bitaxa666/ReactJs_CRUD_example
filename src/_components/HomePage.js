/**
 * Created by user on 3/30/18.
 */
import React, {Component} from 'react';


class HomePage extends Component {


    constructor(props) {
        super(props);
        this.state = {
            title: '' /*//в самом обьекте есть поле, но при тесте оно не активно?*/
        };
    };

    handleChange = (e) => {
        this.setState({title: e.target.value});

    };


    render() {
        return(
            <div className="home_page_class">
                <h1>This is Home Page</h1>
                <div className="ui container">
                    <form>
                       <h1>Home page test</h1>
                        <div className="home-form">
                            <label htmlFor="title">Title</label>
                            <input
                                name="title"
                                value={this.state.title}
                                onChange={this.handleChange}
                                id="title"
                            />
                            <br/>
                            <span>Errors</span><br />
                            <span>Test snapshoot Errors</span>
                        </div>
                    </form>
                </div>
            </div>
        );
    }
}
export default HomePage;