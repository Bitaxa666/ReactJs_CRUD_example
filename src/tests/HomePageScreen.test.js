/**
 * Created by user on 4/4/18.
 */
import React from 'react';
import { configure, shallow, mount, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HomePage from '../_components/HomePage';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });


describe('HomePage', function() {

    it('render main layout', () => {
        const wrapper = shallow(<HomePage />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});