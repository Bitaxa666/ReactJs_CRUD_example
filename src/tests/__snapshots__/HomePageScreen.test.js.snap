// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage render main layout 1`] = `
<div
  className="home_page_class"
>
  <h1>
    This is Home Page
  </h1>
  <div
    className="ui container"
  >
    <form>
      <h1>
        Home page test
      </h1>
      <div
        className="home-form"
      >
        <label
          htmlFor="title"
        >
          Title
        </label>
        <input
          id="title"
          name="title"
          onChange={[Function]}
          value=""
        />
        <br />
        <span>
          Errors
        </span>
        <br />
        <span>
          Test snapshoot Errors
        </span>
      </div>
    </form>
  </div>
</div>
`;
